l3kernel/l3file.dtx

```diff
@@ -1499,10 +1499,21 @@
 %
 % \begin{macro}{\@@_new:N}
 %   As for read streams, copy \tn{newwrite}, making sure
-%   that it is not \tn{outer}.
+%   that it is not \tn{outer}. For \ConTeXt{}, we have to
+%   deal with the fact that \tn{newwrite} works like our
+%   own: it actually checks before altering definition.
 %    \begin{macrocode}
 \exp_args:NNf \cs_new_protected:Npn \@@_new:N
   { \exp_args:NNc \exp_after:wN \exp_stop_f: { newwrite } }
+\cs_if_exist:NT \normalend
```

Comment:
Is there a better (aka more standard test) that we can use? It may not be necessary, but it looks a bit odd I think to test for `\normalend`.

Reply:
This test is already used three times in l3file (four times after this pull request). It may be worth defining `\sys_if_format_context:TF` or similar to make the intent clearer.

Reply:
When I was setting up `l3names` to support ConTeXt, checking for `\normalend` was the way I looked for ConTeXt - so unless there is an obvious better one, this seems reasonable to me

Reply:
ConTeXt defines an entire suite of `\normal...` commands. If we wanted to be extra careful and make sure that the user didn't just happen to define `\normalend`, we could check that all of them are defined. Such a test would be best wrapped into a function `\sys_if_format_context:TF` or similar, because it would be tedious to repeat and prone to inconsistencies.

Reply:
one could, but one could perhaps alternatively test for something which is more internal to context and less likely to be used. But since it never posed a problem so far I guess either is overkill

Reply:
Fixed in https://github.com/latex3/latex3/pull/1114/commits/3c340a01867e5760ca4c713155800adebe55de9b.